### PR TITLE
Mnist dag

### PR DESCRIPTION
**Pull request by ShellyGoel**
Created Reduce and ReLU DagNode and Dot Product function and tested each.

### CODE REVIEW

**Comment by rdaly525** [comment]
@ShellyGoel could you merge this with master?

File: BitFlow/Eval/TorchEval.py [hunk 1]
```diff
@@ -21,8 +21,25 @@ def eval_Round(self, a, prec, node: DagNode):
         scale = 2.0**prec
         return IntRound(a * scale)/scale
 
+    def eval_Relu(self, a, node: DagNode):
```
Comment (rdaly525) [inline BitFlow/Eval/TorchEval.py]: Unfortunately this will not quite work if you are passing in a torch array. Can you do a test for this case? I suspect there is already a torch.relu function somewhere which you can use.

File: BitFlow/Eval/TorchEval.py [hunk 2]
```diff
@@ -21,8 +21,25 @@ def eval_Round(self, a, prec, node: DagNode):
...
+                return 0
+
     def eval_Select(self, a, node: DagNode):
+
         if len(a.shape) == 1:
             return a[node.index]
         else:
             return a[:, node.index]
+
+    def eval_Reduce(self, a, node: DagNode):
```
Comment (rdaly525) [inline BitFlow/Eval/TorchEval.py]: Could you add a parameter to the Reduce DagNode that takes in the dimension you want to sum over? You should be able to just use torch.sum(a, dim=node.reduce_dim) for the evaluation function. Could you also write a couple tests testing reducing over different dimensions?

**Review by rdaly525 — CHANGES_REQUESTED** [review]
